Automattic/woocommerce-payments
# Add Deposits API documentation

Fixes #7839

#### Changes proposed in this Pull Request

This PR adds documentation for the WooPayments Deposits API endpoints `/wp-json/wc/v3/payments/deposits/*`.

> [!note]
> I've uploaded a preview of this PR's version of the docs [here](https://heroic-sprinkles-47a306.netlify.app/#deposits).

- [x] `GET deposits/overview-all`
- [x] `GET deposits/overview`
  * doesn’t honour the `store_currency_is` param, only returns the stripe default currency. [Code ref](https://github.com/Automattic/woocommerce-payments-server/blob/cdbf5cbc0d522ad540fd246f4112ee1507a1c8f1/server/wp-content/rest-api-plugins/endpoints/wcpay/class-deposits-controller.php).
- [x] `GET deposits`
- [x] `GET deposits/summary`
- [x] `GET deposits/(?P<deposit_id>\w+)`
- [x] `POST deposits`
- [x] `POST deposits/download`


These docs are published via the [`deploy-api-docs.yml` GH action](https://github.com/Automattic/woocommerce-payments/actions/workflows/deploy-api-docs.yml) to https://automattic.github.io/woocommerce-payments/ once merged to `develop`.



#### Testing instructions


* Build the docs locally (requires Docker)
  * `cd docs/rest-api`
  * `./build.sh` to build the docs via a Docker container.
  * `npx serve build` to serve the built HTML at `localhost:3000`
* Alternatively, I've uploaded this PR's version of the docs manually [here](https://heroic-sprinkles-47a306.netlify.app/#deposits).
* Find and navigate to the **Deposits** section via the sidebar
* You can verify responses by running the `cURL` commands using the `consumer_key` and `consumer_secret` available from the secret store: `Helix WooExpress WC API key`. I have verified each using Postman, so they _should_ be accurate.



-------------------

- [x] Tested on mobile (or does not apply) N/A - mobile styles handled by Slate docs generator.

**Post merge**
